torch/utils/data/dataset.py

```diff
@@ -25,6 +25,32 @@
 T = TypeVar('T')
 
 
+class DataChunk(object):
```

Comment:
How about we make `DataChunk` inheriting `list`? Then, we only need to handle `isinstance(data, list)` and return `type(data)(result)`

Reply:
Not always, sometimes we need `raw_iterator` for DataFrames